File: lua/raindrops/raindrops.lua
return function (n)
    local factors = {
        { 3, 'Pling' },
        { 5, 'Plang' },
        { 7, 'Plong' }
    }

    local ret = ''
    for _, tbl in ipairs(factors) do
        local factor, noise = unpack(tbl)
        if n % factor == 0 then ret = ret .. noise end
    end
    return ret ~= '' and ret or tostring(n)
end

File: ruby/raindrops/raindrops.rb
module BookKeeping
  VERSION = 1
end

class Raindrops
  class << self
    def convert(n)
      sounds = ""
      sounds << 'Pling' if n % 3 == 0
      sounds << 'Plang' if n % 5 == 0
      sounds << 'Plong' if n % 7 == 0
      sounds.empty? ? n.to_s : sounds
    end
  end
end
